File: 1012 - area/area.kt
import java.util.*

fun areadotrianguloretangulo( baseA : Double, alturaC : Double ) : Double
{
    return ((baseA * alturaC) / 2.0 );
}

fun areadocirculo( raioC : Double ) : Double
{
    return ( 3.14159 * raioC * raioC);
}

fun areadotrapezio( baseA : Double, baseB : Double, alturaC : Double ) : Double
{
    return (((baseA + baseB) * alturaC ) / 2.0 );
}

fun areadoquadrado( ladoB : Double ) : Double
{
    return (ladoB * ladoB);
}

fun areadoretangulo( ladoA : Double, ladoB : Double ) : Double
{
    return (ladoA * ladoB);
}

fun main( args: Array<String>){
    var(a, b, c) = readLine()!!.split(' ');
    println("TRIANGULO: %.3f".format( areadotrianguloretangulo(a.toDouble(), c.toDouble()) ));
    println("CIRCULO: %.3f".format( areadocirculo(c.toDouble()) ));
    println("TRAPEZIO: %.3f".format( areadotrapezio(a.toDouble(), b.toDouble(), c.toDouble()) ));
    println("QUADRADO: %.3f".format( areadoquadrado(b.toDouble()) ));
    println("RETANGULO: %.3f".format( areadoretangulo(a.toDouble(), b.toDouble()) ));
}

File: 1012 - area/area.scala
object Main{

    def areadotrianguloretangulo( baseA: Double, alturaC: Double) : Double =
        ((baseA * alturaC) / 2.0 )

    def areadocirculo( raioC : Double ) : Double =
        ( 3.14159 * raioC * raioC)

    def areadotrapezio( baseA : Double , baseB : Double , alturaC : Double ) : Double =
        (((baseA + baseB) * alturaC ) / 2.0 )

    def areadoquadrado( ladoB : Double ) : Double =
        (ladoB * ladoB)

    def areadoretangulo( ladoA : Double, ladoB : Double ) : Double =
        (ladoA * ladoB)

    def main( args: Array[String]){
        val linha = ( scala.io.StdIn.readLine() ).split(" ")
        var a = linha(0).toDouble
        var b = linha(1).toDouble
        var c = linha(2).toDouble

        printf("TRIANGULO: %.3f\n", areadotrianguloretangulo(a, c) );
        printf("CIRCULO: %.3f\n", areadocirculo(c) );
        printf("TRAPEZIO: %.3f\n", areadotrapezio(a, b, c) );
        printf("QUADRADO: %.3f\n", areadoquadrado(b) );
        printf("RETANGULO: %.3f\n", areadoretangulo(a, b) );
    }
}